data/modules/Rondel/Rondel.lua

```diff
@@ -0,0 +1,225 @@
+
+	if option == -1 then
+		form:Close()
+		return
+	end
+
+	form:SetMessage(l_rondel.SOLFED_INTEL_INTRO)
+
+	if option == 1 then
+		ads[ref] = nil
+		form:RemoveAdvertOnClose()
+		form:SetMessage(l_rondel.PERFECT)
+		Game.player:AddMoney(250000)
+		rondel_prize = true
+		return
+	elseif option == 2 then
+		form:SetMessage(l_rondel.YOU_VISITED_STAR_SYSTEM)
+	elseif option == 3 then
+		form:SetMessage(l_rondel.CREDITS_PROMISE)
+	end
+
+	form:AddOption(l_rondel.YOU_SHALL_HAVE_THE_DATA, 1)
+	form:AddOption(l_rondel.WHAT_IS_THIS_ABOUT, 2)
+	form:AddOption(l_rondel.NOT_INTERESTED, 3)
+
+end
+
+local onDelete = function (ref)
+	ads[ref] = nil
+end
+
+local onCreateBB = function (station)
+	if rondel_prize or not rondel_victory then return end
+
+	local ad = {
+		title    = l_rondel.HEY_OVER_HERE,
+		--message  = "",
+		station  = station,
+		--character = Character.New({armour=false}),
+	}
+
+	local ref = station:AddAdvert({
+		description = ad.title,
+		--icon        = "",
+		onChat      = onChat,
+		onDelete    = onDelete})
+	ads[ref] = ad
+end
+
+local attackShip = function (ship)
+	for i = 1, #patrol do
+		patrol[i]:AIKill(ship)
+	end
+end
+
+local onShipDestroyed = function (ship, attacker)
+	if ship:IsPlayer() or attacker == nil or Game.system.name ~= "Rondel" then return end
+
+	for i = 1, #patrol do
+		if patrol[i] == ship then
+			table.remove(patrol, i)
+			if #patrol > 1 then
+				Comms.ImportantMessage(l_rondel.DEFENSE_CRAFT_ELIMINATED, patrol[1].label)
+			elseif #patrol == 0 then
+				rondel_victory = true
+			end
+			break
+		elseif patrol[i] == attacker then
+			Comms.ImportantMessage(l_rondel.INTERCEPTION_SUCCESSFUL, attacker.label)
+			break
+		end
+	end
+end
+
+local onShipFiring = function (ship)
+	if Game.system.name ~= "Rondel" then return end
+
+	local police = ShipDef[Game.system.faction.policeShip]
+	if ship.shipId ~= police.id then
+		for i = 1, #patrol do
+			if ship:DistanceTo(patrol[i]) <= 4000000 and not shipFiring then
+				shipFiring = true
+				Comms.ImportantMessage(string.interp(l_rondel.WEAPONS_EVASIVE_ACTION, patrol[1].label))
+				attackShip(ship)
+				break
+			end
+		end
+	end
+end
+
+local onJettison = function (ship, cargo)
+	if Game.system.name ~= "Rondel" then return end
+	if not jetissionedCargo and #patrol > 1 then
+		Comms.ImportantMessage(l_rondel.JETTISON_DEFENSIVE_PRECAUTION, patrol[1].label)
+		jetissionedCargo = true
+	end
+	attackShip(ship)
+end
+
+local onEnterSystem = function (player)
+	if not player:IsPlayer() then return end
+
+	local system = Game.system
+	if system.name ~= "Rondel" then return end
+
+	local tolerance = 1
+	local hyperdrive = Game.player:GetEquip('engine',1)
+	if hyperdrive.fuel == cargo.military_fuel then
+		tolerance = 0.5
+	end
+
+	local ship
+	local shipdef = ShipDef[system.faction.policeShip]
+	for i = 1, 7 do
+		ship = Space.SpawnShipNear(shipdef.id, player, 50, 100)
+		ship:SetLabel(l_rondel.HABER_DEFENSE_CRAFT)
+		ship:AddEquip(Equipment.laser.pulsecannon_2mw)
```

Comment:
Sigh... I'm beginning to realize that we need a lib of some sort to handle spawning ships from templates. This works for now though, no need to change it.

Reply:
I'm clicking "resolve conversation" so it stops telling me there are changes requested.